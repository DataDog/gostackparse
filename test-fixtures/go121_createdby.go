//go:build ignore

package main

import (
	"fmt"
	"runtime"
	"sync"
)

func main() {
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		printStack()
	}()
	wg.Wait()
}

func printStack() {
	buf := make([]byte, 1<<20)
	n := runtime.Stack(buf, true)
	fmt.Printf("%s\n", buf[:n])
}
