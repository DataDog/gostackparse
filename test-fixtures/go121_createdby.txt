goroutine 18 [running]:
main.printStack()
	/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go:23 +0x40
main.main.func1()
	/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go:16 +0x4c
created by main.main in goroutine 1
	/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go:14 +0x70

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0x140000021a0?)
	/Users/nick.ripley/repos/go/src/runtime/sema.go:62 +0x2c
sync.(*WaitGroup).Wait(0x1400011a010)
	/Users/nick.ripley/repos/go/src/sync/waitgroup.go:116 +0x74
main.main()
	/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go:18 +0x78