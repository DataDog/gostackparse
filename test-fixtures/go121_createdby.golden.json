{
  "Errors": null,
  "Goroutines": [
    {
      "ID": 18,
      "State": "running",
      "Wait": 0,
      "LockedToThread": false,
      "Stack": [
        {
          "Func": "main.printStack",
          "File": "/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go",
          "Line": 23
        },
        {
          "Func": "main.main.func1",
          "File": "/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go",
          "Line": 16
        }
      ],
      "FramesElided": false,
      "CreatedBy": {
        "Func": "main.main",
        "File": "/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go",
        "Line": 14
      }
    },
    {
      "ID": 1,
      "State": "semacquire",
      "Wait": 0,
      "LockedToThread": false,
      "Stack": [
        {
          "Func": "sync.runtime_Semacquire",
          "File": "/Users/nick.ripley/repos/go/src/runtime/sema.go",
          "Line": 62
        },
        {
          "Func": "sync.(*WaitGroup).Wait",
          "File": "/Users/nick.ripley/repos/go/src/sync/waitgroup.go",
          "Line": 116
        },
        {
          "Func": "main.main",
          "File": "/Users/nick.ripley/repos/gostackparse/test-fixtures/go121_createdby.go",
          "Line": 18
        }
      ],
      "FramesElided": false,
      "CreatedBy": null
    }
  ]
}
